### PR TITLE
Add LastModifiedTime to Status for ScalableTarget and ScalingPolicy resources

### PR DESCRIPTION
**Pull request by mbaijal**
#### Description of changes:
Add LastModifiedTime to Status for ScalableTarget and ScalingPolicy resources
TODO: Timezone of `CreationTimestamp` is different from the `CreationTime` and `LastModifiedTime` 

#### Testing:
 - Manually tested that the field is present in the Status of the kubectl get response and gets modified on resource updates. 
 - Kind tests pass locally

#### Output as shown below:
```
$ kubectl describe scalingpolicy.applicationautoscaling.services.k8s.aws/m-scalingpolicy-may13

Name:         m-scalingpolicy-may13
Namespace:    default
Labels:       <none>
Annotations:  <none>
API Version:  applicationautoscaling.services.k8s.aws/v1alpha1
Kind:         ScalingPolicy
Metadata:
  Creation Timestamp:  2021-09-08T18:35:14Z
  Finalizers:
    finalizers.applicationautoscaling.services.k8s.aws/ScalingPolicy
  Generation:        3
  Resource Version:  2247
  Self Link:         /apis/applicationautoscaling.services.k8s.aws/v1alpha1/namespaces/default/scalingpolicies/m-scalingpolicy-may13
  UID:               <truncated>
Spec:
  Policy Name:         m-scalingpolicy-may13
  Policy Type:         TargetTrackingScaling
  Resource ID:         endpoint/<endpointname>/variant/AllTraffic
  Scalable Dimension:  sagemaker:variant:DesiredInstanceCount
  Service Namespace:   sagemaker
  Target Tracking Scaling Policy Configuration:
    Predefined Metric Specification:
      Predefined Metric Type:  SageMakerVariantInvocationsPerInstance
    Scale In Cooldown:         700
    Scale Out Cooldown:        300
    Target Value:              80
Status:
  Ack Resource Metadata:
    Arn:               arn:aws:autoscaling:<truncated>
    Owner Account ID:  169544399729
  Alarms:
    Alarm ARN:   <truncated>
    Alarm Name:  <truncated>
    Alarm ARN:   <truncated>
    Alarm Name:  <truncated>
  Conditions:
  Creation Time:       2021-09-08T18:35:15Z
  Last Modified Time:  2021-09-08T18:36:31Z
  ```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/scalable_target/custom_api.go [hunk 1]
```diff
@@ -15,7 +15,10 @@ package scalable_target
 
 import (
 	"context"
+	svcapitypes "github.com/aws-controllers-k8s/applicationautoscaling-controller/apis/v1alpha1"
 	svcsdk "github.com/aws/aws-sdk-go/service/applicationautoscaling"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
```
Comment (mbaijal) [inline pkg/resource/scalable_target/custom_api.go:21]: Need reordering

File: pkg/resource/scaling_policy/custom_api.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	svcapitypes "github.com/aws-controllers-k8s/applicationautoscaling-controller/apis/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// customSetOutputUpdate sets the LastModifiedTime field to the current time post an update
+func (rm *resourceManager) customSetLastModifiedTime(ko *svcapitypes.ScalingPolicy) {
+	currentTime := metav1.Time{Time: time.Now()}
```
Comment (ryansteakley) [inline pkg/resource/scaling_policy/custom_api.go:24]: Will this time.now() local time be the same as the time zone shown in creationTime?

File: pkg/resource/scalable_target/custom_api.go [hunk 3]
```diff
@@ -31,3 +34,9 @@ func (rm *resourceManager) customDescribeScalableTarget(
 		input.SetResourceIds(resourceIDList)
 	}
 }
+
+// customSetOutputUpdate sets the LastModifiedTime field to the current time post an update
+func (rm *resourceManager) customSetLastModifiedTime(ko *svcapitypes.ScalableTarget) {
+	currentTime := metav1.Time{Time: time.Now()}
```
Comment (jaypipes) [inline pkg/resource/scalable_target/custom_api.go:40]: Here and below, please use [`time.Now().UTC()`](https://pkg.go.dev/time#Time.UTC)

**Review by jaypipes — CHANGES_REQUESTED** [review]
Should be UTC since CreationTime is in UTC

File: pkg/resource/scalable_target/custom_api.go (hunk 3 above)
Comment (mbaijal) [reply pkg/resource/scalable_target/custom_api.go:40]: yes I noticed that later - thank you!

Comment (mbaijal) [reply pkg/resource/scalable_target/custom_api.go]: done.

File: pkg/resource/scaling_policy/custom_api.go (hunk 2 above)
Comment (mbaijal) [reply pkg/resource/scaling_policy/custom_api.go]: They should all be in UTC now

**Review by jaypipes — APPROVED** [review]
:+1:

**Review by surajkota — COMMENTED** [review]
2 things

Do you think it's better to set the modified time = creation time post create instead of current time post api response?

Can you add an assert to integ tests. Notebook lifecycle config has a similar assert

**Pull request merged**